### PR TITLE
Visitor can see a list of products

### PR DESCRIPTION
**Pull request by vick3d**
## Feature: Visitor can see a list of products
https://www.pivotaltracker.com/story/show/163901539

### User Story:
* As a visitor,
In order to see what food is available
I would like to be able to see a menu

### Description of PR:
- Creation of product controller and product model
- Add factory bot for creating products
- Tests created for showing products

### CODE REVIEW

**Review by NiciP — APPROVED** [review]

File: app/controllers/product_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class ProductController < ApplicationController
+  
+  def index
+    @products = Product.all
+  end
+
```
Comment (LauraRe) [inline app/controllers/product_controller.rb]: remove space

File: app/views/product/index.html.haml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+%h1 Product#index
+
```
Comment (LauraRe) [inline app/views/product/index.html.haml]: remove space

File: features/step_definitions/basic_steps.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  
+  When("I am on the {string} page") do |page|
+    case page
+    when "product" 
+      visit product_index_path
+    else 
+      return false 
+    end
+  end
+
```
Comment (LauraRe) [inline features/step_definitions/basic_steps.rb]: remove spaces at the end of the file

File: features/step_definitions/assertion_steps.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+Then("I should see {string}") do |content|
+    expect(page).to have_content content
+  end
+
```
Comment (LauraRe) [inline features/step_definitions/assertion_steps.rb:12]: remove space

File: features/visitor_can_see_a_list_of_products.feature [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        Then I should see "Pizza"
+        And I should see "Garlic bread"
+        And I should see "Chocolate"
+        And I should see "10 SEK"
+        And I should see "spicy"
+        And I should see "3 SEK"
+        And I should see "stinky"
+        And I should see "5 SEK"
+        And I should see "fattening"
+
```
Comment (LauraRe) [inline features/visitor_can_see_a_list_of_products.feature]: remove space

**Review by LauraRe — CHANGES_REQUESTED** [review]
Remove extra spaces. See details in files.

**Review by LauraRe — APPROVED** [review]
Looks good now.

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -1,2 +1,4 @@
 class ApplicationController < ActionController::Base
+    def index
```
Comment (oliverochman) [inline app/controllers/application_controller.rb]: Why is this here?

File: app/controllers/product_controller.rb (hunk 1 above)
Comment (vick3d) [reply app/controllers/product_controller.rb]: deleted

File: features/step_definitions/basic_steps.rb (hunk 3 above)
Comment (vick3d) [reply features/step_definitions/basic_steps.rb]: deleted

File: features/step_definitions/assertion_steps.rb (hunk 4 above)
Comment (vick3d) [reply features/step_definitions/assertion_steps.rb:12]: deleted

File: features/visitor_can_see_a_list_of_products.feature (hunk 5 above)
Comment (vick3d) [reply features/visitor_can_see_a_list_of_products.feature]: deleted

File: app/models/product.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class Product < ApplicationRecord
+    validates :product_name, presence: true
```
Comment (oliverochman) [inline app/models/product.rb]: refactor this to `validates_presence_of :product_name, :price, :description`

File: config/routes.rb [hunk 8]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get 'product/index'
+  resources :products
+  root 'product#index'
```
Comment (oliverochman) [inline config/routes.rb]: refactor this to 
```
root controller: :products, action: :index
resources :products, only: [:index]

File: db/migrate/20190219090835_create_products.rb [hunk 9]
```diff
@@ -0,0 +1,11 @@
+class CreateProducts < ActiveRecord::Migration[5.2]
+  def change
+    create_table :products do |t|
+      t.string :product_name
```
Comment (oliverochman) [inline db/migrate/20190219090835_create_products.rb:4]: Do you have any reason why you don't just call this `name`?

File: app/controllers/product_controller.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+class ProductController < ApplicationController
```
Comment (oliverochman) [inline app/controllers/product_controller.rb]: This needs to be renamed to `ProductsController`

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: db/migrate/20190219090835_create_products.rb (hunk 9 above)
Comment (vick3d) [reply db/migrate/20190219090835_create_products.rb:4]: We might have another table with name later on. So this is just more clear.

File: app/models/product.rb (hunk 7 above)
Comment (vick3d) [reply app/models/product.rb]: Changed to
```
Rails.application.routes.draw do
  get 'product/index'
  root controller: :products, action: :index
  resources :products, only: [:index]
end
```

Comment (oliverochman) [reply app/models/product.rb]: You dont need `get 'product/index'`

File: db/migrate/20190219090835_create_products.rb (hunk 9 above)
Comment (oliverochman) [reply db/migrate/20190219090835_create_products.rb:4]: The models are independent of each other, so there is no need to give each model unique attribute names. When calling on the products name it will look silly (e.g `Product.product_name`)

File: app/controllers/application_controller.rb (hunk 6 above)
Comment (vick3d) [reply app/controllers/application_controller.rb]: we accidentally added an index method to this one instead of products controller.
Forgot to delete the method. It's gone now

File: app/models/product.rb (hunk 7 above)
Comment (vick3d) [reply app/models/product.rb]: fixed

File: config/routes.rb (hunk 8 above)
Comment (vick3d) [reply config/routes.rb]: fixed

File: db/migrate/20190219090835_create_products.rb (hunk 9 above)
Comment (vick3d) [reply db/migrate/20190219090835_create_products.rb:4]: fixed

File: app/controllers/product_controller.rb (hunk 10 above)
Comment (vick3d) [reply app/controllers/product_controller.rb]: fixed

**Review by oliverochman — APPROVED** [review]
I need to see more git ponging in the future guys!
